lint/rule.go

```diff
@@ -108,6 +114,10 @@ type FieldRule struct {
 	// Problems it finds.
 	LintField func(*desc.FieldDescriptor) []Problem
 
+	// OnlyIf accepts a FieldDescriptor and determines whether this rule
+	// is applicable.
+	OnlyIf func(*desc.FieldDescriptor) bool
```

Comment:
My feeling is that checking message is more common -- but it is easy to get the message. Since we are not using this function now, maybe we should postpone it.

Reply:
The `OnlyIf` function can easily call `f.GetMessage()` in that case.

Reply:
Fair enough. Thanks!